### PR TITLE
Diverging colors

### PR DESCRIPTION
**Pull request by nanodan**
Added the middle swatch to the diverging colors (for 7 base) so that when odd n is selected the middle swatch stays constant as it does on colorbrewer2.org and modified code for selecting less than n colors for sequential or diverging or qualitative so that results mimic colorbrewer2.org more closely.  Also allows for n<7 qualitative through truncation.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@BibMartin do you have some time to look at this one #23 ?

File: branca/utilities.py [hunk 1]
```diff
@@ -164,61 +170,97 @@ def color_brewer(color_code, n=6):
...
+                           '#f4cae4', '#e6f5c9', '#fff2ae','#f1e2cc'],
                'Set1': ['#e41a1c', '#377eb8', '#4daf4a',
-                        '#984ea3', '#ff7f00', '#ffff33'],
+                        '#984ea3', '#ff7f00', '#ffff33','#a65628'],
                'Set2': ['#66c2a5', '#fc8d62', '#8da0cb',
-                        '#e78ac3', '#a6d854', '#ffd92f'],
+                        '#e78ac3', '#a6d854', '#ffd92f','#e5c494'],
                'Set3': ['#8dd3c7', '#ffffb3', '#bebada',
-                        '#fb8072', '#80b1d3', '#fdb462'],
+                        '#fb8072', '#80b1d3', '#fdb462','#b3de69'],
```
Comment (ocefpaf) [inline branca/utilities.py]: I wonder if we should factor out the colormaps to another file to make them easier to manage...

Comment (nanodan) [reply branca/utilities.py]: I think that makes sense as well. Also the way colorbrewer works you have different end-points for color that jump between 5 to 7 and then again at 11 so if we want to follow it 100% we will need a few different list definitions depending on desired length, or we can just pick one.

**Comment by ocefpaf** [comment]
Sorry, this one fly under the radar. @nanodan do you mind resolving the conflicts?

@BibMartin if you have time can you take a quick look? If not I'll review this again over the weekend.

**Comment by nanodan** [comment]
I'll resolve the conflicts this week so you can review this weekend.

**Comment by ocefpaf** [comment]
Thanks! Sorry for the delay :grimacing:

**Comment by nanodan** [comment]
Sorry haven't had time yet - will try to finish this weekend ASAP.

**Comment by ocefpaf** [comment]
No problem ; -)

**Comment by nanodan** [comment]
Had a bunch of messed up conflicts - just easier to sync my master and create a new branch and I'll submit a new pull request.

**Comment by nanodan** [comment]
Okay new pull request ready.

**Pull request closed without merging**